src/main/java/com/neuvector/Scanner.java

```diff
@@ -329,6 +339,64 @@ private static String maskCredential(String message, String credential){
         return message.replace(credential, "******");
     }
 
+    static String[] getDockerGroupCmdArgs(String scanReportPath) {
+        String[] cmdGroupArgs = {"", ""};
+        String UID = executeCommand("id -u");
+        if (!scanResultsFileExist(scanReportPath) || !isRootFile(scanReportPath) && (UID!= null && !UID.isEmpty())) {
```

Comment:
With the latest changes you have, `isRootFile(scanReportPath)` can be `null` and must be checked for.. 

This if statement is getting quite large and difficult to read, maybe we can break it.. 

Also we need a better name for `rootFile`.. It is not a `rootFile`.. It is owned by the `rootUser`.. Something similar to:

```
    static String[] getDockerGroupCmdArgs(String scanReportPath) {
        String[] cmdGroupArgs = {"", ""};

        // no user arg if file exists and it is owned by root
        Boolean ownedByRoot = isRootFile(scanReportPath);
        if (scanResultsFileExist(scanReportPath) && ownedByRoot != null && ownedByRoot) {
            return cmdGroupArgs;
        }

        String userId = executeCommand("id -u");
        if (userId!= null && !userId.isEmpty()) {
            cmdGroupArgs[0] = "-u";
            cmdGroupArgs[1] = userId;
        }

        return cmdGroupArgs;
    }
```

Reply:
5f5f2cecaf55f0f205133425202e8781bd064377